### PR TITLE
Merge WIP nitric run into develop

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: pkg/cmd/run/root.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+			MinWorkers: 0,
+			MaxWorkers: 100,
+		})
+
+		// Start a new gateway plugin
+		gw, err := run.NewGateway()
+
+		if err != nil {
+			cobra.CheckErr(err)
+		}
```
Comment (asalkeld) [inline pkg/cmd/run/root.go]:
```suggestion
		cobra.CheckErr(err)
```
no point wrapping this in an err != nil check as this is partly what CheckErr() does

Comment (tjholm) [reply pkg/cmd/run/root.go]: fixed.

File: pkg/cmd/run/root.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		memerr := make(chan error)
+		go func(errch chan error) {
+			errch <- mem.Start()
+		}(memerr)
+
+		time.Sleep(time.Second * time.Duration(2))
+
+		functions, err := run.FunctionsFromHandlers(ctx, files)
+		cobra.CheckErr(err)
+		if err != nil {
```
Comment (davemooreuws) [inline pkg/cmd/run/root.go]: left over if condition, remove

File: pkg/cmd/stack/root.go [hunk 3]
```diff
@@ -94,24 +113,69 @@ var stackCreateCmd = &cobra.Command{
...
 	Args: cobra.MaximumNArgs(2),
 }
 
+var stackDescribeCmd = &cobra.Command{
+	Use:   "describe [handler pattern]",
+	Short: "describe stack dependencies",
+	Long:  `Describes stack dependencies`,
+	Run: func(cmd *cobra.Command, args []string) {
+		stackPath, err := filepath.Abs(stack.StackPath())
+		if err != nil {
```
Comment (davemooreuws) [inline pkg/cmd/stack/root.go]: remove if statement

**Review by davemooreuws — CHANGES_REQUESTED** [review]

File: pkg/cmd/stack/root.go (hunk 3 above)
Comment (tjholm) [reply pkg/cmd/stack/root.go]: fixed.

File: pkg/cmd/run/root.go (hunk 2 above)
Comment (tjholm) [reply pkg/cmd/run/root.go]: fixed.

**Review by davemooreuws — APPROVED** [review]

**Comment by davemooreuws** [comment]
Approved, linting needs to be fixed as a final change.

**Pull request merged**